### PR TITLE
Fermichem 2.6 - Adds the HPLC: a new roundstart method of detecting purity, as well as a method of partially purifying reagents

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

Adds the High-performance liquid chromatography machine - a new roundstart method of detecting purity, as well as a method of partially purifying reagents.
A single machine has been added to each of the 4(5) maps - this lets people determine their purity based off the size of the green (pure portion) to the orange (impure portion) of a reagent's peak shown on the mas spectroscopy display. If a reagent is impure - it will be displayed as red. 
In addition to this function, the HPLC can purify reagents by selecting a range of them and pressing the purify button. This will cost some time depending on the mass of the reagent and will purify it up it's default purity (in the tweaked reagent cases - 75%, for the reagents I've yet to get to - 100%). It will also reduce the volume accordingly (so in essence you are reducing the volume of the impure parts).
The sprite itself will indicate when it's running, so you don't need to be nearby it or use the UI to know when it's done
Example below!

https://user-images.githubusercontent.com/33956696/111876962-f433c580-8998-11eb-9676-e725cf4c8420.mp4

## To do:
Add induvidial masses to reagents - but I can do that while it's being reviewed

## Why It's Good For The Game

Currently the only ways to detect purity is with the handheld chem analyser, upgrading the heater and purity tester reagents- but most of the feedback I've had is about how purity is a pain for people. I'm hoping this will help solve that without making the analyser obsolete.

Also, unless you're intending to purposely cause havock then low purity batches are usually thrown out - this way you can recouperate some of your failed reaction volume instead.

## Changelog
:cl:
adds: Adds the HPLC purifying machine - this machine will indicate the purity of the reagent in it's input beaker and can partially purify selected reagents
adds: Gives each reagent a mass
fix: Fixes the debug recipe machine runtiming when hit by a beaker
code: Adds alt_click_secondary() which is called when right clicking on things with alt held down
imageadd: added some more icons for the HPLC
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
slick UI 👍

**Comment by Farquaar** [comment]
Really freaking cool

File: tgui/packages/tgui/interfaces/MassSpec.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+            beaker={beaker1Contents} />
+          {!!beaker1Contents.length && (
+            <Box>
+              {"Eta of selection: " + round(eta) + " seconds"}
+            </Box>
+          )}
+        </Section>
+        <Section
+          title="Output beaker"
+          buttons={!!beaker2Contents && (
```
Comment (tralezab) [inline tgui/packages/tgui/interfaces/MassSpec.js:93]: add a fill prop here and you can have your tgui interface not have dead space (i mean the section will still be empty, but it'll look a pinch nicer :)

**Comment by tralezab** [comment]
ehh i was gonna comment something on the tgui but actually nevermind it's pretty much perfect lmao

**Pull request merged**